route/build/src/io_rpointfile.f90

```diff
@@ -18,20 +19,33 @@ MODULE io_rpointfile
   ! *********************************************************************
   ! public subroutine: main routine to define new output file
   ! *********************************************************************
-  SUBROUTINE io_rpfile(mode, ierr, message)
+  SUBROUTINE io_rpfile(mode, ierr, message, curDatetime)
 
     implicit none
     ! argument variables
-    character(1),   intent(in)     :: mode        ! io mode: 'r' or 'w'
-    integer(i4b),   intent(out)    :: ierr        ! error code
-    character(*),   intent(out)    :: message     ! error message
+    character(1),   intent(in)           :: mode          ! io mode: 'r' or 'w'
+    integer(i4b),   intent(out)          :: ierr          ! error code
+    character(*),   intent(out)          :: message       ! error message
+    type(datetime), intent(in), optional :: curDatetime   ! current datetime
+    integer                              :: sec_in_day    ! time in second
+    character(len=17)                    :: timestamp     ! datetime string in yyyy-mm-dd-sssss
+    character(len=strLen)                :: rpntfil_path  ! rpointer file path
 
     ierr=0; message='io_rpfile/'
 
+    if (present(curDatetime)) then
+      sec_in_day = curDatetime%hour()*60*60+curDatetime%minute()*60+nint(curDatetime%sec())
```

Comment:
Change 60*60 to nint(secphour) and 60 to nint(secpmin) in public_vars

Reply:
fixed.